src/inmanta/loader.py

```diff
@@ -167,9 +167,16 @@ def get_types(self) -> Iterable[Tuple[str, List[SourceInfo]]]:
 @dataclass(frozen=True)
 class ModuleSource:
     name: str
-    source: str
+    source: bytes
     hash_value: str
 
+    def is_bytecode(self) -> bool:
+        try:
+            self.source.decode("utf-8")
```

Comment:
I was thinking, would it not be easier to use the magic number to recognize the file. 

It turns out it is now, because the magic changes often. 

https://github.com/python/cpython/blob/main/Lib/importlib/_bootstrap_external.py#L233

Perhaps add a comment here to prevent this same investigation from being done again.